nifi-nar-bundles/nifi-network-bundle/nifi-network-processors/src/main/java/org/apache/nifi/processors/network/ParseNetflowv5.java

```diff
@@ -0,0 +1,258 @@
+        @WritesAttribute(attribute = "netflowv5.record.*", description = "The key and value generated by the parsing of the record fields.") })
+
+public class ParseNetflowv5 extends AbstractProcessor {
+    private String destination;
+    // Add mapper
+    private static final ObjectMapper mapper = new ObjectMapper();
+
+    public static final String DESTINATION_CONTENT = "flowfile-content";
+    public static final String DESTINATION_ATTRIBUTES = "flowfile-attribute";
+    public static final PropertyDescriptor FIELDS_DESTINATION = new PropertyDescriptor.Builder().name("FIELDS_DESTINATION").displayName("Parsed fields destination")
+            .description("Indicates whether the results of the parser are written " + "to the FlowFile content or a FlowFile attribute; if using " + DESTINATION_ATTRIBUTES
+                    + ", fields will be populated as attributes. If set to " + DESTINATION_CONTENT + ", the netflowv5 field will be converted into a flat JSON object.")
+            .required(true).allowableValues(DESTINATION_CONTENT, DESTINATION_ATTRIBUTES).defaultValue(DESTINATION_CONTENT).build();
+
+    public static final Relationship REL_FAILURE = new Relationship.Builder().name("failure")
+            .description("Any FlowFile that could not be parsed as a netflowv5 message will be transferred to this Relationship without any attributes being added").build();
+    public static final Relationship REL_ORIGINAL = new Relationship.Builder().name("original").description("The original raw content").build();
+    public static final Relationship REL_SUCCESS = new Relationship.Builder().name("success")
+            .description("Any FlowFile that is successfully parsed as a netflowv5 data will be transferred to this Relationship.").build();
+
+    public static final List<PropertyDescriptor> PROPERTIES = Collections.unmodifiableList(Arrays.asList(FIELDS_DESTINATION));
+    public static final Set<Relationship> RELATIONSHIPS = Collections.unmodifiableSet(new HashSet<>(Arrays.asList(REL_FAILURE, REL_ORIGINAL, REL_SUCCESS)));
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return RELATIONSHIPS;
+    }
+
+    @Override
+    public final List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return PROPERTIES;
+    }
+
+    @OnScheduled
+    public void onScheduled(final ProcessContext context) {
+        destination = context.getProperty(FIELDS_DESTINATION).getValue();
+    }
+
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSession session) throws ProcessException {
+        FlowFile flowFile = session.get();
+        if (flowFile == null) {
+            return;
+        }
+
+        final OptionalInt portNumber = resolvePort(flowFile);
+        final Netflowv5Parser parser = new Netflowv5Parser(portNumber);
+
+        final byte[] buffer = new byte[(int) flowFile.getSize()];
```

Comment:
This can be very dangerous. How big are the flowfiles going to be on average?

Reply:
As far as i know, netflow packets are generally exported as UDP datagrams. So, I think the buffer length should also falls under UDP datagram limit..

Reply:
@MikeThomsen  Can you share your view on this?

Reply:
According to Wikipedia, it's a maximum of 64kb, so I'm good with that.